### PR TITLE
Optimize away read optimization during client registration

### PR DESCRIPTION
**Pull request by larshesel**
Need to bench this (if it is even correct)

### CODE REVIEW

File: apps/vmq_server/src/vmq_reg_mgr.erl [hunk 1]
```diff
@@ -162,15 +162,14 @@ setup_queue(SubscriberId, Subs, Acc) ->
...
+        {delete, _SubscriberId, _} ->
             %% TODO: we might consider a queue cleanup here.
             ignore;
         {update, _SubscriberId, [], []} ->
             %% noop
             ignore;
-        {update, SubscriberId, _, _} ->
-            Subs = vmq_reg:subscriptions_for_subscriber_id(SubscriberId),
-            handle_new_sub_event(SubscriberId, Subs);
+        {update, SubscriberId, _OldSubs, NewSubs} ->
```
Comment (dergraf) [inline apps/vmq_server/src/vmq_reg_mgr.erl]: Does this work? The `_OldSubs` and `NewSubs` don't contain the 'real' DB object, they contain the changes (from `vmq_subscriber:get_changes`) made to them, e.g. what was added or what was removed... Or in other words the `NewSubs` isn't identical to what you have in `Subs`.

Comment (dergraf) [reply apps/vmq_server/src/vmq_reg_mgr.erl]: Ah, missed the fact that you've removed the call to `get_changed` ;)

File: apps/vmq_server/src/vmq_reg_trie.erl [hunk 2]
```diff
@@ -239,9 +239,11 @@ code_change(_OldVsn, State, _Extra) ->
 %%%===================================================================
 handle_event(Handler, Event) ->
     case Handler(Event) of
-        {delete, SubscriberId, Subs} ->
-            vmq_subscriber:fold(fun handle_delete_event/2, SubscriberId, Subs);
-        {update, SubscriberId, ToRemove, ToAdd} ->
+        {delete, SubscriberId, Subscriptions} ->
+            Removed = vmq_subscriber:get_changes(Subscriptions),
```
[the remark this answers is not in the document]
Comment (larshesel) [reply apps/vmq_server/src/vmq_reg_trie.erl]: `vmq_subscriber:get_changes/2` returns a `{ToRemove, ToAdd}`, but `vmq_subscriber:get_changes/1` just returns the subscriptions in the argument, so I think the above is correct.

[the remark this answers is not in the document]
Comment (larshesel) [reply apps/vmq_server/src/vmq_reg_trie.erl]: In fact, the `vmq_subscriber:get_changes/1` doesn't do anything at this point, it just returns the argument, so we could completely remove it.

**Comment by larshesel** [comment]
> I don't think specific benchmarks are required as the improvement is quite obvious (saving additional database roundtrips)

I still think some microbenchmarking would be interesting, just to get a feel for the impact this has.

**Pull request merged**